packages/upswyng-server/src/routes/api/weather.ts

```diff
@@ -0,0 +1,113 @@
+import axios from "axios";
+
+const getCurrentWeather = async (
+  lat: number | null,
+  lon: number | null
+): Promise<TWeatherCurrentResponse> => {
+  try {
+    const { data } = await axios.get<TWeatherCurrentResponse>(
+      "https://api.openweathermap.org/data/2.5/weather?units=imperial",
+      {
+        params: {
+          lat: lat,
+          lon: lon,
+          APPID: process.env.REACT_APP_OPEN_WEATHER_API_KEY,
+        },
+      }
+    );
+    return data;
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+class WeatherCache {
+  millisecondsToExp: number;
+  fetchFunction: typeof getCurrentWeather;
+  cache: TWeatherCurrentResponse;
+  fetchDate: Date;
+  constructor(fetchFunction: () => Promise<TWeatherCurrentResponse>) {
+    this.millisecondsToExp = 600000;
+    this.fetchFunction = fetchFunction;
+    this.cache = null;
+    this.getData = this.getData.bind(this);
+    this.resetCache = this.resetCache.bind(this);
+    this.isCacheExpired = this.isCacheExpired.bind(this);
+    this.fetchDate = new Date(0);
+  }
+  async getData() {
+    if (!this.cache || this.isCacheExpired()) {
+      console.log("Fetching new weather data");
+      const data = await this.fetchFunction(40.015, -105.2705);
+      this.cache = data;
+      this.fetchDate = new Date();
+      return data;
+    } else {
+      console.log("Returning cached weather data");
+      return Promise.resolve(this.cache);
+    }
+  }
+  isCacheExpired() {
+    return (
+      this.fetchDate.getTime() + this.millisecondsToExp < new Date().getTime()
+    );
+  }
+  resetCache() {
+    this.fetchDate = new Date(0);
+  }
+}
+
+const weather = new WeatherCache(() => getCurrentWeather(40.015, -105.2705));
+
+export async function get(req, res) {
+  try {
+    const result = await weather.getData();
+    res.writeHead(200, { "Content-Type": "application/json" });
+    return res.end(JSON.stringify({ ...result }));
+  } catch (e) {
+    console.error(e);
+    res.writeHead(500, {
+      "Content-Type": "application/json",
+    });
+
+    return res.end(
+      JSON.stringify({
+        message: `Error fetching weather information: ${e.message}`,
+      })
+    );
```

Comment:
can shorten to:
```
return res.setStatus(500).json({message: blah blah});
```
(`json` will automatically apply the `Content-Type` and stringify for you) (I just learned all this..we can refactor some endpoints)

Reply:
res.status 😄